### DISCUSSION TITLE
Option to retain original page dimensions?

### DISCUSSION THREAD

**Discussion by someone-i-know** [ocrmypdf/OCRmyPDF#1178] [Q&A]
I tried OCR for a pdf in which each page is roughly A4 dimensions.. The OCRed file has large portions of empty space around the text, increasing the page width and height. Is there a way to retain the original page height and width while performing OCR?

I cannot show the pdf but this is what it looks like: (left is OCRed, right is original)

![git screenshot](https://github.com/ocrmypdf/OCRmyPDF/assets/149052161/41ec88ee-adeb-407d-be9b-279ae6ef478a)

**Comment by jbarlow83** [comment]
Generally speaking ocrmypdf preserves original page.

This issue concerns a specific PDF file and I will need that file to test it.

If the file contains private information, here are some options:

- You can [encrypt the file with my public key](https://github.com/ocrmypdf/OCRmyPDF/wiki) as described here, to ensure that only @jbarlow83 can open it.
- You can attempt to produce a non-private file that likely has a similar issue. For example, if the PDF that caused the problem was created with a specific scanner software, perhaps that software is the problem. Scan some other document or a redacted document with the same equipment and software. (If you are not sure what to scan, print a page from Wikipedia.)
- You could try using Acrobat's Redact tools, or similar tools in other PDF editors, to remove personal information. Note that this will change the PDF and might fix the issue.
- You could run `qpdf --json yourfile.pdf > yourfile.pdf.json` and upload the resulting .json file here. That contains a structural description of the contents of the PDF. In most cases, this will not help; but occasionally it does.

If you cannot share the file, you could try the following workarounds to produce a visually similar PDF:

- Refrying the PDF using Ghostscript `gs -q -sDEVICE=pdfwrite -o out.pdf in.pdf`.
- Printing the file to a "Print to PDF" virtual printer, which will reconstruct the PDF and perhaps create a new PDF without the issue.
- Using `ocrmypdf --force-ocr` which is more aggressive about removing document contents.

If you still cannot share the file after the options above, please close the issue, because there is no way I will be able to reproduce it if you cannot provide a way for me to test it.

**Reply by someone-i-know** [reply]
Thanks for responding. I have extracted one page from the file which exhibits this same behaviour.
I hope this helps to identify the problem.

[to my mother.pdf](https://github.com/ocrmypdf/OCRmyPDF/files/13175644/to.my.mother.pdf)

Command used:
ocrmypdf --force-ocr [input path] [output path]

**Reply by jbarlow83** [reply]
This is a bug - will address soon

**Reply by jbarlow83** [reply]
#1181
